plugins/obs-ffmpeg/obs-ffmpeg-source.c

```diff
@@ -288,6 +298,45 @@ static void ffmpeg_source_open(struct ffmpeg_source *s)
 	}
 }
 
+static void ffmpeg_source_start(struct ffmpeg_source *s)
+{
+	if (!s->media_valid)
+		ffmpeg_source_open(s);
+
+	if (s->media_valid) {
+		mp_media_play(&s->media, s->is_looping);
+		if (s->is_local_file)
+			obs_source_show_preloaded_video(s->source);
+		set_media_state(s, OBS_MEDIA_STATE_PLAYING);
+		obs_source_media_started(s->source);
+	}
+}
+
+static void *ffpmeg_source_reconnect(void *data)
+{
+	struct ffmpeg_source *s = data;
+	os_sleep_ms(s->reconnect_delay_time * 1000);
+	FF_BLOG(LOG_INFO, "MP: start reconnect thread");
+	do {
+		if (s->stop_reconect) {
+			break;
+		}
+
+		if (s->media_valid) {
+			break;
+		}
```

Comment:
Suggestion from @RytoEX:

```suggestion
		if (s->media_valid)
			break;
```